### ISSUE TITLE
Eigen data type issue, build failure

### ISSUE THREAD

**Issue by brlcad** [bmpix/PolyVectorization#2]
Hi!  I've ran into a compilation error with the latest that I haven't been able to work out, and I'm hoping you can help me get a successful compile.  Here's the error:

```code
agua:.build morrison$ make
/Users/morrison/Applications/bin/cmake -S/Users/morrison/src/PolyVectorization -B/Users/morrison/src/PolyVectorization/.build --check-build-system CMakeFiles/Makefile.cmake 0
/Users/morrison/Applications/bin/cmake -E cmake_progress_start /Users/morrison/src/PolyVectorization/.build/CMakeFiles /Users/morrison/src/PolyVectorization/.build/CMakeFiles/progress.marks
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/Makefile2 all
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/polyvector_thing_autogen.dir/build.make CMakeFiles/polyvector_thing_autogen.dir/depend
cd /Users/morrison/src/PolyVectorization/.build && /Users/morrison/Applications/bin/cmake -E cmake_depends "Unix Makefiles" /Users/morrison/src/PolyVectorization /Users/morrison/src/PolyVectorization /Users/morrison/src/PolyVectorization/.build /Users/morrison/src/PolyVectorization/.build /Users/morrison/src/PolyVectorization/.build/CMakeFiles/polyvector_thing_autogen.dir/DependInfo.cmake --color=
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/polyvector_thing_autogen.dir/build.make CMakeFiles/polyvector_thing_autogen.dir/build
[  2%] Automatic MOC and UIC for target polyvector_thing
/Users/morrison/Applications/bin/cmake -E cmake_autogen /Users/morrison/src/PolyVectorization/.build/CMakeFiles/polyvector_thing_autogen.dir/AutogenInfo.cmake ""
[  2%] Built target polyvector_thing_autogen
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/polyvector_thing.dir/build.make CMakeFiles/polyvector_thing.dir/depend
cd /Users/morrison/src/PolyVectorization/.build && /Users/morrison/Applications/bin/cmake -E cmake_depends "Unix Makefiles" /Users/morrison/src/PolyVectorization /Users/morrison/src/PolyVectorization /Users/morrison/src/PolyVectorization/.build /Users/morrison/src/PolyVectorization/.build /Users/morrison/src/PolyVectorization/.build/CMakeFiles/polyvector_thing.dir/DependInfo.cmake --color=
Scanning dependencies of target polyvector_thing
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f CMakeFiles/polyvector_thing.dir/build.make CMakeFiles/polyvector_thing.dir/build
[  5%] Building CXX object CMakeFiles/polyvector_thing.dir/src/bilinearInterpCoeff.cpp.o
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++  -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DWITH_GUI=1 -I/Users/morrison/src/PolyVectorization/.build -I/Users/morrison/src/PolyVectorization -I/Users/morrison/src/PolyVectorization/.build/polyvector_thing_autogen/include -I/Users/morrison/src/PolyVectorization/src -I/usr/local/include -iframework /Users/morrison/Applications/Qt-5.13/5.13.0/clang_64/lib -isystem /Users/morrison/Applications/Qt-5.13/5.13.0/clang_64/lib/QtCore.framework/Headers -isystem /Users/morrison/Applications/Qt-5.13/5.13.0/clang_64/./mkspecs/macx-clang -isystem /Users/morrison/Applications/Qt-5.13/5.13.0/clang_64/lib/QtWidgets.framework/Headers -isystem /Users/morrison/Applications/Qt-5.13/5.13.0/clang_64/lib/QtGui.framework/Headers -isystem /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/OpenGL.framework/Headers -isystem /usr/local/include/opencv4 -isystem /usr/local/include/eigen3  -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk   -fPIC -std=gnu++11 -o CMakeFiles/polyvector_thing.dir/src/bilinearInterpCoeff.cpp.o -c /Users/morrison/src/PolyVectorization/src/bilinearInterpCoeff.cpp
/Users/morrison/src/PolyVectorization/src/bilinearInterpCoeff.cpp:32:21: error: no viable conversion from 'const Product<Eigen::Product<Eigen::CwiseBinaryOp<Eigen::internal::scalar_product_op<double, double>, const Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, const
      Eigen::Matrix<double, 1, 2, 1, 1, 2> >, const Eigen::Transpose<Eigen::Matrix<double, 2, 1, 0, 2, 1> > >, Eigen::Matrix<std::__1::complex<double>, 2, 2, 0, 2, 2>, 0>, Eigen::Matrix<double, 2, 1, 0, 2, 1> >' to 'std::__1::complex<double>'
                result[rootIdx] = (1.0 / ((i2 - i1)*(j2 - j1))) * Eigen::Vector2d(i2 - i, i - i1).transpose() * M * Eigen::Vector2d(j2 - j, j - j1);
                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/complex:381:28: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'const
      Product<Eigen::Product<Eigen::CwiseBinaryOp<Eigen::internal::scalar_product_op<double, double>, const Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, const Eigen::Matrix<double, 1, 2, 1, 1, 2> >, const Eigen::Transpose<Eigen::Matrix<double, 2, 1, 0, 2, 1> > >,
      Eigen::Matrix<std::__1::complex<double>, 2, 2, 0, 2, 2>, 0>, Eigen::Matrix<double, 2, 1, 0, 2, 1> >' to 'const std::__1::complex<double> &' for 1st argument
class _LIBCPP_TEMPLATE_VIS complex<double>
                           ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/complex:381:28: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'const
      Product<Eigen::Product<Eigen::CwiseBinaryOp<Eigen::internal::scalar_product_op<double, double>, const Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, const Eigen::Matrix<double, 1, 2, 1, 1, 2> >, const Eigen::Transpose<Eigen::Matrix<double, 2, 1, 0, 2, 1> > >,
      Eigen::Matrix<std::__1::complex<double>, 2, 2, 0, 2, 2>, 0>, Eigen::Matrix<double, 2, 1, 0, 2, 1> >' to 'std::__1::complex<double> &&' for 1st argument
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/complex:388:49: note: candidate constructor not viable: no known conversion from 'const Product<Eigen::Product<Eigen::CwiseBinaryOp<Eigen::internal::scalar_product_op<double, double>, const
      Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, const Eigen::Matrix<double, 1, 2, 1, 1, 2> >, const Eigen::Transpose<Eigen::Matrix<double, 2, 1, 0, 2, 1> > >, Eigen::Matrix<std::__1::complex<double>, 2, 2, 0, 2, 2>, 0>, Eigen::Matrix<double, 2, 1, 0, 2, 1> >' to 'double'
      for 1st argument
    _LIBCPP_INLINE_VISIBILITY _LIBCPP_CONSTEXPR complex(double __re = 0.0, double __im = 0.0)
                                                ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/complex:508:18: note: candidate constructor not viable: no known conversion from 'const Product<Eigen::Product<Eigen::CwiseBinaryOp<Eigen::internal::scalar_product_op<double, double>, const
      Eigen::CwiseNullaryOp<Eigen::internal::scalar_constant_op<double>, const Eigen::Matrix<double, 1, 2, 1, 1, 2> >, const Eigen::Transpose<Eigen::Matrix<double, 2, 1, 0, 2, 1> > >, Eigen::Matrix<std::__1::complex<double>, 2, 2, 0, 2, 2>, 0>, Eigen::Matrix<double, 2, 1, 0, 2, 1> >' to
      'const complex<float> &' for 1st argument
complex<double>::complex(const complex<float>& __c)
                 ^
/usr/local/include/eigen3/Eigen/src/Core/Product.h:130:23: note: candidate function
  EIGEN_STRONG_INLINE operator const Scalar() const
                      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/complex:381:28: note: passing argument to parameter here
class _LIBCPP_TEMPLATE_VIS complex<double>
                           ^
1 error generated.
make[2]: *** [CMakeFiles/polyvector_thing.dir/src/bilinearInterpCoeff.cpp.o] Error 1
make[1]: *** [CMakeFiles/polyvector_thing.dir/all] Error 2
make: *** [all] Error 2
```

I'm not familiar with Eigen, but I think it's complaining about multiplying a std::complex with a Eigen::Matrix2cd ?  Or perhaps it's a constness issue and a cast somewhere would get me going ?  I'm not sure what to do about it.

**Comment by brlcad** [comment]
For what it's worth, I commented out everything after the parenthetical, and I got a successful compile (with Qt enabled after other edits).  That is, `result[rootIdx] = (1.0 / ((i2 - i1)*(j2 - j1)));`

Undoubtedly totally breaks something, but the application did chug along and amazingly completed successfully after 3100 iterations on my test image.  Now it's "Computing Reeb graph..." on 5273 curves.

Any idea what the proper edit for that line should be?  I hopefully didn't introduce an infinite loop!
